anomalib/models/dfkde/model.py

```diff
@@ -20,13 +20,71 @@
 import torchvision
 from omegaconf.dictconfig import DictConfig
 from omegaconf.listconfig import ListConfig
-from torch import Tensor
+from torch import Tensor, nn
 
 from anomalib.models.components import AnomalyModule, FeatureExtractor
 
 from .normality_model import NormalityModel
 
 
+class DfkdeModel(nn.Module):
+    """DFKDR model.
+
+    Args:
+        backbone (nn.Module): Feature extraction backbone.
+        filter_count (int): Number of filters.
+        threshold_steepness (float): Threshold steepness for normality model.
+        threshold_offset (float): Threshold offset for normality model.
+    """
+
+    def __init__(
+        self, backbone: nn.Module, filter_count: int, threshold_steepness: float, threshold_offset: float
+    ) -> None:
+        super().__init__()
+        self.feature_extractor = FeatureExtractor(backbone=backbone(pretrained=True), layers=["avgpool"]).eval()
+
+        self.normality_model = NormalityModel(
+            filter_count=filter_count,
+            threshold_steepness=threshold_steepness,
+            threshold_offset=threshold_offset,
+        )
+
+    def get_features(self, batch: Tensor) -> Tensor:
+        """Extract features from the pretrained network.
+
+        Args:
+            batch (Tensor): Image batch.
+
+        Returns:
+            Tensor: Tensor containing extracted features.
+        """
+        self.feature_extractor.eval()
+        layer_outputs = self.feature_extractor(batch)
+        layer_outputs = torch.cat(list(layer_outputs.values())).detach()
+        return layer_outputs
+
+    def fit_normality_model(self, embeddings: List[Tensor]):
```

Comment:
i would prefer this is called just `fit`. During its usage, it would just be `self.model.fit(embeddings`